wrangler-transform/src/main/java/io/cdap/wrangler/WranglerErrorUtil.java

```diff
@@ -0,0 +1,123 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import io.cdap.cdap.api.exception.ErrorCategory;
+import io.cdap.cdap.api.exception.ErrorType;
+import io.cdap.cdap.api.exception.ErrorUtils;
+import io.cdap.cdap.api.exception.ProgramFailureException;
+import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveLoadException;
+import io.cdap.wrangler.api.DirectiveNotFoundException;
+import io.cdap.wrangler.api.DirectiveParseException;
+import io.cdap.wrangler.api.RecipeException;
+import io.cdap.wrangler.expression.ELException;
+import io.cdap.wrangler.utils.RecordConvertorException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Error util file to handle exceptions caught in Wrangler plugin
+ */
+public final class WranglerErrorUtil {
+
+
+  private static final Map<String, String> TERMINAL_EXCEPTIONS = ImmutableMap.<String, String>builder()
+      .put(DirectiveParseException.class.getName(), "Parsing-Directive")
+      .put(PreconditionException.class.getName(), "Precondition")
+      .put(DirectiveExecutionException.class.getName(), "Executing-Directive")
+      .put(DirectiveLoadException.class.getName(), "Loading-Directive")
+      .put(DirectiveNotFoundException.class.getName(), "Directive-Not-Found")
+      .put(RecordConvertorException.class.getName(), "Record-Conversion")
+      .put(ELException.class.getName(), "ExpressionLanguage-Parsing").build();
+
+  private static final Map<String, String> NON_TERMINAL_EXCEPTIONS = ImmutableMap.<String, String>builder()
+      .put(RecipeException.class.getName(), "Executing-Recipe").build();
+
+  /**
+   * Private constructor to prevent instantiation of this utility class.
+   * <p>
+   * This class is designed to contain only static utility methods for handling exceptions and
+   * should not be instantiated. Any attempt to create an instance of this class will result in an
+   * {@link IllegalStateException}.
+   */
+  private WranglerErrorUtil() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  /**
+   * Traverses the causal chain of the given Throwable to find specific exceptions. If a terminal
+   * exception is found, it returns a corresponding ProgramFailureException. If a non-terminal
+   * exception is found, it is stored as a fallback. Otherwise, a generic ProgramFailureException is
+   * returned.
+   *
+   * @param e            the Throwable to analyze
+   * @param errorMessage default error message if no terminal exception is found
+   * @param errorType    the error type to categorize the failure
+   * @return a ProgramFailureException with specific or generic error details
+   */
+  public static ProgramFailureException getProgramFailureExceptionDetailsFromChain(Throwable e,
+      String errorMessage, String errorReason, ErrorType errorType) {
```

Comment:
At all other places we are taking `errorReason` before `errorMessage`. Please keep it consistent.